packages/api/cms-api/src/content-generation/openai-content-generation.service.ts

```diff
@@ -2,12 +2,23 @@ import { AzureKeyCredential, ChatRequestMessage, OpenAIClient } from "@azure/ope
 import { Injectable } from "@nestjs/common";
 import fetch from "node-fetch";
 
-import { OpenAiContentGenerationConfig } from "./content-generation.config";
-import { ContentGenerationServiceInterface } from "./content-generation.service.interface";
+import { ContentGenerationServiceInterface } from "./content-generation-service.interface";
+
+export type OpenAiContentGenerationConfig<T> = {
+    [K in keyof T]: {
+        deploymentId: string;
+        apiKey: string;
+        apiUrl: string;
+    };
+};
 
 @Injectable()
 export class OpenAiContentGenerationService implements ContentGenerationServiceInterface {
-    constructor(private readonly config: OpenAiContentGenerationConfig<ContentGenerationServiceInterface>) {}
+    config: OpenAiContentGenerationConfig<ContentGenerationServiceInterface>;
+
+    init(config: OpenAiContentGenerationConfig<ContentGenerationServiceInterface>): void {
+        this.config = config;
+    }
```

Comment:
Alternatively the OpenAiContentGenerationService couldn't implement the ContentGenerationServiceInterface and just pass the API credentials when calling `generateAltText`. This would also allow passing custom model specific options, for instance, `maxTokens`

Reply:
I made following changes:

- using the global config object in ContentGenerationService (instead of looping the config through the module)
- pass API config directly to the methods as you suggested
- add imports to the module to allow importing other modules to have access to services in the app

Can you check again?